cat /usr/include/linux/version.h
#define RHEL_MAJOR 8
#define RHEL_MINOR 4
RTLNX_RHEL_MIN(8,4)

--- ./VirtualBox-kmod-6.1.16/vboxsf/vfsmod.c.orig	2020-12-13 18:57:48.614216635 +0000
+++ ./VirtualBox-kmod-6.1.16/vboxsf/vfsmod.c	2020-12-13 18:59:41.537387526 +0000
@@ -45,7 +45,7 @@
 #include "version-generated.h"
 #include "revision-generated.h"
 #include "product-generated.h"
-#if RTLNX_VER_MIN(5,0,0)
+#if RTLNX_VER_MIN(5,0,0) || RTLNX_RHEL_MIN(8,4)
 # include <uapi/linux/mount.h> /* for MS_REMOUNT */
 #elif RTLNX_VER_MAX(3,3,0)
 # include <linux/mount.h>
